processor/src/main/java/org/mapstruct/ap/MappingProcessor.java

```diff
@@ -198,19 +201,20 @@ public SourceVersion getSupportedSourceVersion() {
         return SourceVersion.latestSupported();
     }
 
+    @NotNull
     @Override
-    public boolean process(final Set<? extends TypeElement> annotations, final RoundEnvironment roundEnvironment) {
+    public Set<XElement> process(@NotNull XProcessingEnv processingEnv, @NotNull Map<String, ? extends Set<? extends XElement>> elementsByAnnotation, boolean isLastRound) {
```

Comment:
the processingEnv was a field in the superclass before, I was not sure if it stays the same object during the whole lifetime so I added it as a parameter in several locations, would have to check if field works as well